cleanup_delete.go

```diff
@@ -152,5 +124,29 @@ func (dq *deleteQueue) start(ctx context.Context) {
 				dq.deleteFromPending()
 			}
 		}
+		select {
+		case <-ctx.Done():
+			dq.drain()
+			// drain the delete queue and return
+			go func() {
+				for {
```

Comment:
Yep, but as it's part of the shutdown process I don't care if it leaks 🙉

The drain func uses a timeout so there may be later sends to this channel that would block, which is what the leaked routine is for.